packages/styles/gulpfile.js

```diff
@@ -160,6 +160,48 @@ gulp.task(
   }),
 );
 
+/**
+ * Get all available components names from the components package and add them to the scss file (packages\styles\src\utilities\_not-defined.scss) which sets initial visibility to hidden (for unregistered state).
+ */
+
+gulp.task('generate-not-defined-components-scss', done => {
+  const filePath = path.join(__dirname, '../components/src/index.ts');
+  fs.readFile(filePath, 'utf8', (err, data) => {
+    if (err) {
+      done(err);
+      return;
+    }
+
+    const content = data;
+    const componentNames = content
+      .match(/export \{ (\w+) \} from/g)
+      .map(match => match.replace(/export \{ (\w+) \} from/, '$1'));
+
+    const kebabCaseNames = componentNames.map(name =>
+      name.replace(/([a-z])([A-Z])/g, '$1-$2').toLowerCase(),
+    );
+
+    const componentNamesContent = `/*
+  Initial visibility of the components is set to hidden to prevent 'flickering' effect due to stencil js/scss delay.
+*/
+
+:where(
+    ${kebabCaseNames.join(',\n    ')}
+):not(:defined) {
+  visibility: hidden;
+}
+    `;
```

Comment:
Would be nice, if we could extract this code and save it (for example) as a `_not-defined.template.scss`.
We can then load it in this gulp task and update it with theg given kebabCaseNames, but keep the format and have it checked in to git. While the output file would not be checked in (because it's getting generated).